### DISCUSSION TITLE
MFEM - joule miniapp with coil.gen mesh

### DISCUSSION THREAD

**Discussion by dhk27182** [mfem/mfem#2023] [Q&A] [answered]
Hello MFEM-Team,

I'm interested in learning about FEM implementations, especially for eddy current electromagnetic problems. When trying to
run the joule miniapp with the coil.gen mesh, it aborts, if I start it with more than 2 slots (I'm on a 6 core machine):

```
$ mpirun -np 4 joule -m coil.gen  -p coil
....
Skin depth sqrt(2.0/(wj*mj*sj)) = 0.551329
Skin depth sqrt(2.0*dt/(mj*sj)) = 0.126157



Verification failed: (stria_flag[i] == stria_master_flag[i]) is false:
 --> inconsistent vertex ordering found, shared triangle 539: (6899, 6898, 3171), local flag: 2, master flag: 1
 ... in function: virtual void mfem::ParMesh::ReorientTetMesh()
 ... in file: mesh/pmesh.cpp:2726
```
With only 2 slots (-np 2), the computation works. 
Is there something, that I am missing ?

Best regards,
dhk27182

**Comment by mlstowell** [comment, marked as answer]
Hi, @dhk27182 ,

Using the latest version of MFEM's master branch I ran the above command line using 2 to 10 processors.  I obtained essentially the same error message when using 6 or 8 processors.  All other cases ran fine.

I also used the `mesh-explorer` miniapp to convert `coil.gen` to MFEM's native mesh format (using the 'S' option from the `mesh-explorer` menu).  Using this version of the mesh `joule` ran fine for all processor configurations from 2 to 10.

Clearly we have a problem.  It may just be in the mesh reader for this mesh type or it could be within the `mfem::Mesh` class, although this seems unlikely.  Further investigation will be required to find the cause.

Thanks, @dhk27182 , for bringing this to our attention.

Best wishes,
Mark

**Reply by dhk27182** [reply]
Hello mlstowell,

thank you for the fast reply. With your hint to converting the mesh format with `mesh-explorer` I was able to successfully run the computation with 6 cores.

And even nicer, if I could help you to improve the software with my question.

Best regards,
dhk27182
